crates/stackable-operator/src/builder/pod/probe.rs

```diff
@@ -0,0 +1,330 @@
+use std::num::TryFromIntError;
+
+use k8s_openapi::{
+    api::core::v1::{ExecAction, GRPCAction, HTTPGetAction, Probe, TCPSocketAction},
+    apimachinery::pkg::util::intstr::IntOrString,
+};
+use snafu::{ResultExt, Snafu, ensure};
+
+use crate::time::Duration;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display(
+        "The probe's {field:?} duration of {duration} is too long, as it's seconds doesn't fit into an i32"
+    ))]
+    DurationTooLong {
+        source: TryFromIntError,
+        field: String,
+        duration: Duration,
+    },
+
+    #[snafu(display("The probe period is zero, but it needs to be a positive duration"))]
+    PeriodIsZero {},
+}
+
+#[derive(Debug)]
+pub struct ProbeBuilder<Action, Period> {
```

Comment:
I do think we should have docs on pub items.

It doesn't have to be super comprehensive, and examples can be added to the module level docs.

```suggestion
/// Valid Kubernetes probe builder
///
/// The upstream struct [NameOfStruct] does not prevent invalid probe configurations
/// which leads to surprises at runtime which can be deeply hidden.
#[derive(Debug)]
pub struct ProbeBuilder<Action, Period> {
```

Reply:
No opinion, moved to module docs in fd5a099